song/song_server.py

```diff
@@ -89,10 +94,16 @@ def beat_handler(self, _, note):
                 self.song_machine.release_lock()
                 self._send_partinfo_to_display()
 
-    def _send_level(self, level):
-        self.osculator_client.send_message('/rack', (level / 10))
-        self.osculator_client.send_message('/osc_notes', (level + 90, 100, 1.0))
-        self.osculator_client.send_message('/osc_notes', (level + 90, 100, 0.0))
+    def send_FX(self):
+        print("chain: {}\nctrl_val: {}".format(self.tonality.chain, self.tonality.ctrl_val))
+        self.osculator_client.send_message('/rack', self.tonality.chain[0])
+        self.osculator_client.send_message('/control{}'.format(self.tonality.chain[1]),  self.tonality.ctrl_val)
+
+    def send_quittung(self, note, velo, dura):
+        self.osculator_client.send_message('/quittung', (note, velo, 1.0))
+        time.sleep(dura)
```

Comment:
Using `time.sleep` in production code is dangerous, in our case even harmful: `sleep` stops the entire process, which in our case implies that new utterances will not be processed while the process is in `sleep`. We should discuss if this is really needed here and, if so, use `threading.Timer` instead of `sleep`

Reply:
versucht..